File: src/main/java/my/study/repository/design/pattern/chain/java/AbstractHandler.java
package my.study.repository.design.pattern.chain.java;

/**
 * @author djh on  2018/9/16 20:22
 * @E-Mail 1544579459@qq.com
 */
abstract class AbstractHandler {

    AbstractHandler mNextHandler;

    /**
     * Handler the request.
     *
     * @param money the request para.
     */
    abstract void handlerRequest(int money);
}

File: src/main/java/my/study/repository/design/pattern/chain/kotlin/AbstractHandler.kt
package my.study.repository.design.pattern.chain.kotlin

/**
 * @author djh on  2018/10/4 18:20
 * @E-Mail 1544579459@qq.com
 */
abstract class AbstractHandler {
    lateinit var nextHandler: AbstractHandler

    abstract fun handlerRequest(money: Int)
}